[BUG] BootKernel error message is hard to read

**Describe the bug**

notice the error
```
* 2023-09-06 20:49:43 (294) [ERROR] Unable to determine running and boot kernels due to the following error:
{"msg":"The running kernel version (3.10.0-1160.88.1.el7.x86_64) does not match that of\nthe default boot entry (3.10.0-1160.95.1.el7.x86_64). This could be due to the kernel\nbeing changed by an update, meaning that a reboot should resolve this.\nHowever, this also could indicate that the system does not have control\nover which kernel and early boot environment (initrd) is used upon\nreboot, which is required to upgrade the operating system with this\nscript.\n\nIf this message remains after a reboot, your server may have been\nconfigured to boot into a particular kernel directly rather than to an\ninstance of the GRUB2 boot loader. This often happens to virtualized\nservers, but physical servers also can have this problem under certain\nconfigurations. Your provider may have a solution to allow booting into\nGRUB2; contact them for further information.\n","id":"Elevate::Blockers::BootKernel::__ANON__"}
```

**To Reproduce**

update the kernel during yum update, then start elevate without rebooting

it's coming from Elevate::Blockers::BootKernel

--- a/elevate-cpanel
+++ b/elevate-cpanel
@@ -386,6 +386,8 @@ BEGIN {    # Suppress load of all of these at earliest point.
     use Cpanel::Kernel::Status ();
 
     # use Cpanel::Exception      ();
+    use Cpanel::YAML ();
+    use Cpanel::JSON ();
 
     use Try::Tiny;
 
@@ -414,10 +416,31 @@ BEGIN {    # Suppress load of all of these at earliest point.
         }
         catch {
             my $ex = $_;
-            $self->has_blocker( "Unable to determine running and boot kernels due to the following error:\n" . Cpanel::Exception::get_string($ex) );
+            $self->has_blocker(
+                "Unable to determine running and boot kernels due to the following error:\n"    #
+                  . _to_str($ex)
+            );
         };
 
         return $ok ? 1 : 0;
+    }
+
+    sub _to_str ($e) {
+        $e //= '';
+
+        my $str = Cpanel::Exception::get_string($e);
+
+        if ( length $str ) {
+
+            my $hash = eval { Cpanel::YAML::Load($str) }    # parse yaml
+              // eval { Cpanel::JSON::Load($str) }          # or json output... we cannot predict
+              // {};
+            if ( ref $hash eq 'HASH' && $hash->{msg} ) {
+                $str = $hash->{msg};
+            }
+        }
+
+        return $str;
     }
 
     1;

--- a/lib/Elevate/Blockers/BootKernel.pm
+++ b/lib/Elevate/Blockers/BootKernel.pm
@@ -21,6 +21,8 @@ use Elevate::Constants ();
 
 use Cpanel::Kernel::Status ();
 use Cpanel::Exception      ();
+use Cpanel::YAML           ();
+use Cpanel::JSON           ();
 
 use Try::Tiny;
 
@@ -49,10 +51,32 @@ sub check ($self) {
     }
     catch {
         my $ex = $_;
-        $self->has_blocker( "Unable to determine running and boot kernels due to the following error:\n" . Cpanel::Exception::get_string($ex) );
+        $self->has_blocker(
+            "Unable to determine running and boot kernels due to the following error:\n"    #
+              . _to_str($ex)
+        );
     };
 
     return $ok ? 1 : 0;
+}
+
+sub _to_str ($e) {
+    $e //= '';
+
+    my $str = Cpanel::Exception::get_string($e);
+
+    if ( length $str ) {
+
+        # can return a YAML or JSON object... handle both
+        my $hash = eval { Cpanel::YAML::Load($str) }    # parse yaml
+          // eval { Cpanel::JSON::Load($str) }          # or json output... we cannot predict
+          // {};
+        if ( ref $hash eq 'HASH' && $hash->{msg} ) {
+            $str = $hash->{msg};
+        }
+    }
+
+    return $str;
 }
 
 1;

